rust/lance/src/dataset.rs

```diff
@@ -440,9 +439,25 @@ impl Dataset {
                         ..Default::default()
                     },
                 )?;
+                let use_v2 = matches!(
+                    d.schema()
+                        .metadata
+                        .get("lance:use_lance_v2")
+                        .map(|s| s.as_str()),
+                    Some("true")
+                );
+                params.use_legacy_format = !use_v2;
+            }
+        } else {
+            if !params.use_legacy_format {
+                schema
+                    .metadata
+                    .insert("lance:use_lance_v2".to_string(), "true".to_string());
```

Comment:
@wjones127 I'm using the schema metadata to store a sort of "soft feature flag".  This is potentially dangerous (users can change the dataset schema and erase this) but since v2 support is only experimental this seems ok.

On the other hand we could use a feature flag but this flag isn't going to be permanent (once v2 is stable we will no longer need it) and so I wasn't sure if we wanted to waste a bit space in our feature flags (though I suppose we could "return it to the pool" once we're done with v2).

Let me know if you would prefer a feature flag.

Reply:
On second thought, I've changed the PR to use feature flags.  It makes sense.  If anyone were writing an independent lance table reader then this would be a significant feature change that would make the resulting tables incompatible.